ui/src/components/DataTable/index.tsx

```diff
@@ -138,12 +148,31 @@ function DataTable<TData, TValue>({
           {showFiltering && (
             <Input
               placeholder={t("Search") + "..."}
-              value={globalFilter ?? ""}
-              onChange={(e) => setGlobalFilter(String(e.target.value))}
+              value={searchTerm ?? globalFilter}
+              onChange={(e) => {
+                const value = String(e.target.value);
+                if (setSearchTerm) {
+                  setSearchTerm(value);
+                }
+                setGlobalFilter(value);
+              }}
```

Comment:
I feel like search term and global filter shouldn't be mixed. In my mind they should be able to work together (unless I'm misunderstanding what global filter would be). 
Just like with the gql having searchTerm and order, etc separated, is there any way here you can keep them working independently? 🙏🏻